src/common/tests/btree.c

```diff
@@ -126,7 +126,6 @@ ik_rec_free(struct btr_instance *tins, struct btr_record *rec, void *args)
 		umem_id_t *rec_ret = (umem_id_t *) args;
 		 /** Provide the buffer to user */
 		*rec_ret	= rec->rec_mmid;
-		rec->rec_mmid	= UMMID_NULL;
```

Comment:
Is it redundant or any side-effect? As my understand, if the caller offer the @args parameter, then it needs to free the space (pointed by rec->rec_mmid) by itself. The @rec will be freed after ik_rec_free() returned. So whether reset the rec_mmid or not will not cause correctness issue, right?

Reply:
Right, the caller is responsible to free the parameter as returned.

Reply:
Then what is the side-effect if we keep this line? Just redundant or some some potential wrong behavior?

Reply:
It seems to me that it's redundant, and it probably may cause warning in some code checking tool?

Reply:
It causes valgrind's pmemcheck tool to report a write to persistent memory that isn't part of the transaction